contracts/oracles/BinanceOracle.sol

```diff
@@ -3,31 +3,54 @@ pragma solidity 0.8.13;
 
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
-import "../interfaces/FeedRegistryInterface.sol";
 import "../interfaces/VBep20Interface.sol";
+import "../interfaces/SIDRegistryInterface.sol";
+import "../interfaces/FeedRegistryInterface.sol";
+import "../interfaces/PublicResolverInterface.sol";
 
 contract BinanceOracle is Initializable {
-    FeedRegistryInterface public feedRegistry;
+    address public sidRegistryAddress;
 
     /// @notice vBNB address
     /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
     address public immutable vBnb;
 
+    /// @notice VAI address
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable vai;
+
     /// @notice Constructor for the implementation contract. Sets immutable variables.
     /// @param vBnbAddress The address of the vBNB
+    /// @param vaiAddress The address of the VAI
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(address vBnbAddress) {
-        require(vBnbAddress != address(0), "can't be zero address");
+    constructor(address vBnbAddress, address vaiAddress) {
+        require(vBnbAddress != address(0), "vBNB can't be zero address");
+        require(vaiAddress != address(0), "VAI can't be zero address");
         vBnb = vBnbAddress;
+        vai = vaiAddress;
         _disableInitializers();
     }
 
     /**
      * @notice Sets the contracts required to fetch prices
-     * @param feed Address of binance oracle feed registry.
+     * @param _sidRegistryAddress Address of SID registry
+     */
+    function initialize(address _sidRegistryAddress) public initializer {
+        sidRegistryAddress = _sidRegistryAddress;
+    }
+
+    /**
+     * @notice Uses Space ID to fetch the feed registry address
+     * @return feedRegistryAddress Address of binance oracle feed registry.
      */
-    function initialize(FeedRegistryInterface feed) public initializer {
-        feedRegistry = feed;
+    function getFeedRegistryAddress() public view returns (address) {
+        bytes32 nodeHash = 0x94fe3821e0768eb35012484db4df61890f9a6ca5bfa984ef8ff717e73139faff;
```

Comment:
is this `nodeHash` same for all networks?

Reply:
Yes. You can find out more here: https://oracle.binance.com/docs/knowledge-base/space-id/